### PR TITLE
Tight bounds clipping max/mins values fixed & 'yAxisUseDecimal' prop added

### PR DESCRIPTION
**Pull request by a-tamang**
Commit #1: My line graphs were being clipped near the min/max regions. Looking at the source code, the problem lied with using Math.round. Fixed it by using Math.floor and Math.ceil for the min and max, respectively.

Commit #2: Graphs with very small fluctuation ( < 1 ) were very confusing to look at without decimals. Added prop 'yAxisUseDecimal' which is false, by default. Enabling it as 'true' shows the decimal points for yAxisLabels

### CODE REVIEW

**Comment by tomauty** [comment]
This looks great

File: src/yAxis.js [hunk 1]
```diff
@@ -46,7 +47,11 @@ export default class YAxis extends Component<void, any, any> {
 		}
 		minBound = (minBound < 0) ? 0 : minBound;
 		let label = minBound + (maxBound - minBound) / this.props.verticalGridStep * index;
-		label = Math.round(label);
+
+		if (this.props.yAxisUseDecimal) {
+			label = Math.round(label);
+		}
```
Comment (tomauty) [inline src/yAxis.js:53]: Wouldn't it be the inverse? i.e. `if (!this.props.yAxisUseDecimal) label = Math.round(label)` ?

**Comment by tomauty** [comment]
I left a comment in there, maybe I'm reading it incorrectly.

**Comment by a-tamang** [comment]
Nope, you're right. It was an error from copy pasting, haha. I've fixed it now.

**Comment by tomauty** [comment]
Cool, I'll do one last review and merge today.

**Comment by a-tamang** [comment]
:+1

**Comment by a-tamang** [comment]
👍

**Comment by tomauty** [comment]
I almost wonder, after looking at this — would it make more sense to just remove the property and make this always true? I'm not sure what the general expectation is from a charting library, but one can always round their own dataset.

**Comment by a-tamang** [comment]
From my understanding, the y-axis labeling is independent of the dataset, is it not? I'm not sure how the y-axis labelling works.

Personally, I'd like to have decimals when I'm dealing with small numbers to depict minute differences whereas for larger numbers, I would think decimals on the y-axis would actually hinder the graph.

**Comment by tomauty** [comment]
Good call.

**Pull request merged**